### PR TITLE
Migrate Schrodinger/sqllogic

### LINKED ISSUE #67: Migrate Schrodinger/sqllogic
## Feature Request

**Is your feature request related to a problem? Please describe:**

**Describe the feature you'd like:**

Refs: https://github.com/pingcap/schrodinger-test/tree/master/sqllogic
Nemesis: randomly

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by mahjonp**
Signed-off-by: mahjonp <junpeng.man@gmail.com>

### What problem does this PR solve? 

close #67 

### What is changed and how does it work?

What's more, I add a `-purge` flag which control whether delete ns when test success, it's useful on CI.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

```log
2020/03/10 14:15:29 sqllogic.go:761: [info] [sqllogic] [run]total 9032129 cases, failed 0, cost 7506 seconds, qps 1203, start 2020-03-10 12:10:23.849019011 +0000 UTC m=+70.345562324, now 2020-03-10 14:15:29.882169445 +0000 UTC m=+7576.378712829

2020/03/10 14:15:32 sqllogic.go:761: [info] [sqllogic] [run]total 9032139 cases, failed 0, cost 7509 seconds, qps 1202, start 2020-03-10 12:10:23.849019011 +0000 UTC m=+70.345562324, now 2020-03-10 14:15:32.882117959 +0000 UTC m=+7579.378661133

2020/03/10 14:15:35 sqllogic.go:761: [info] [sqllogic] [run]total 9032150 cases, failed 0, cost 7512 seconds, qps 1202, start 2020-03-10 12:10:23.849019011 +0000 UTC m=+70.345562324, now 2020-03-10 14:15:35.88212067 +0000 UTC m=+7582.378663914

2020/03/10 14:15:36 sqllogic.go:733: [info] [sqllogic] sqllogictest finished!
2020/03/10 14:15:36 sqllogic.go:761: [info] [sqllogic] [final]total 9032152 cases, failed 0, cost 7513 seconds, qps 1202, start 2020-03-10 12:10:23.849019011 +0000 UTC m=+70.345562324, now 2020-03-10 14:15:36.58838243 +0000 UTC m=+7583.084925884
```

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cluster/k8s.go [hunk 1]
```diff
@@ -41,9 +39,9 @@ func NewK8sProvisioner() (clusterTypes.Provisioner, error) {
 func (k *K8sProvisioner) SetUp(ctx context.Context, spec interface{}) ([]clusterTypes.Node, []clusterTypes.ClientNode, error) {
 	switch s := spec.(type) {
 	case *tidb.TiDBClusterRecommendation:
-		return k.setUpTiDBCluster(ctx, s)
+		return k.setUpTiDBCluster(s)
 	case *binlog.ClusterRecommendation:
-		return k.setUpBinlogCluster(ctx, s)
+		return k.setUpBinlogCluster(s)
```
Comment (you06) [inline pkg/cluster/k8s.go]: why remote context here

File: util/util.go [hunk 2]
```diff
@@ -37,3 +55,58 @@ func RunWithRetry(ctx context.Context, retryCnt int, interval time.Duration, f f
...
+	os.MkdirAll(dest, 0755)
+	err = exec.CommandContext(ctx, "wget", "--tries", "20", "--waitretry", "60",
+		"--retry-connrefused", "--dns-timeout", "60", "--connect-timeout", "60",
+		"--read-timeout", "60", "--directory-prefix", dest, rawURL).Run()
+	return filePath, errors.Annotatef(err, "wget %s to %s", rawURL, dest)
+}
+
+// InstallArchive downloads the URL and extracts the archive to the dest diretory.
+// Supports zip, and tarball.
+func InstallArchive(ctx context.Context, rawURL string, dest string) error {
```
Comment (you06) [inline util/util.go]: What's the purpose of this function? Do we need to download some package?

File: pkg/cluster/k8s.go (hunk 1 above)
Comment (mahjonp) [reply pkg/cluster/k8s.go]: ctx isn't used now, and the process `setUpBinlogCluster` may last a long time especially when lack of resources, so I think using `fixture.Context.WaitClusterReadyDuration` to control the wait time for cluster ready is better.

File: util/util.go (hunk 2 above)
Comment (mahjonp) [reply util/util.go]: Yeah, Sqllogic test need to download a cases archive.

**Pull request merged**